rotkehlchen/api/v1/resources.py

```diff
@@ -2421,6 +2422,33 @@ def get(self, name: str, location: Location) -> Response:
         return self.rest_api.get_user_binance_pairs(name=name, location=location)
 
 
+class BinanceSavingsResource(BaseMethodView):
+
+    def make_post_schema(self) -> BinanceSavingsSchema:
+        with self.rest_api.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            settings = self.rest_api.rotkehlchen.data.db.get_settings(cursor)
+        return BinanceSavingsSchema(
+            treat_eth2_as_eth=settings.treat_eth2_as_eth,
```

Comment:
As I said before why is this needed here? Just cause you are extending the superclass which needs it? In that case then this should be a big hint for you. That class inheritance scheme is wrong.

Reply:
The use case of the inheritance is to handle `eth2 == eth` which is valid imo. Unless I'm not seeing the right thing, I don't think it's wrong.